src/mount.ts

```diff
@@ -42,35 +47,127 @@ interface MountingOptions<Props> {
   shallow?: boolean
 }
 
-// TODO improve the typings of the overloads
-
-type ExtractComponent<T> = T extends { new (): infer PublicInstance }
-  ? PublicInstance
-  : any
+export type ComputedOptions = Record<
+  string,
+  ((ctx?: any) => any) | WritableComputedOptions<any>
+>
+export type ObjectEmitsOptions = Record<
+  string,
+  ((...args: any[]) => any) | null
+>
+export type EmitsOptions = ObjectEmitsOptions | string[]
 
 // Component declared with defineComponent
 export function mount<TestedComponent extends ComponentPublicInstance>(
   originalComponent: { new (): TestedComponent } & Component,
-  options?: MountingOptions<TestedComponent['$props']>
+  options?: MountingOptions<TestedComponent['$props'], TestedComponent['$data']>
 ): VueWrapper<TestedComponent>
-// Component declared with { props: { ... } }
-export function mount<TestedComponent extends ComponentOptionsWithObjectProps>(
-  originalComponent: TestedComponent,
-  options?: MountingOptions<ExtractPropTypes<TestedComponent['props'], false>>
-): VueWrapper<ExtractComponent<TestedComponent>>
-// Component declared with { props: [] }
-export function mount<TestedComponent extends ComponentOptionsWithArrayProps>(
-  originalComponent: TestedComponent,
-  options?: MountingOptions<Record<string, any>>
-): VueWrapper<ExtractComponent<TestedComponent>>
+
+// Functional Component declared with (props, ctx)=> render
+export function mount<Props, RawBindings = object>(
+  setup: (
+    props: Readonly<Props>,
+    ctx: SetupContext
+  ) => RawBindings | RenderFunction,
+
+  options: MountingOptions<Props, {}>
+): VueWrapper<
+  ComponentPublicInstance<
+    Props,
+    RawBindings,
+    {},
+    {},
+    {},
+    // public props
+    VNodeProps & Props
+  >
```

Comment:
:+1: for handling functional component

OOC: do we need to declare all the generics here? If we can achieve what we want with just `ComponentPublicInstance<Props>`? If so, I vote to keep it simple.
If we can't, I would love  to:
- have a comment to explain why
- have a tsd test that fails without it

Reply:
the functional overload can be removed in favour of the 

```ts
// Functional component
export function mount<
  TestedComponent extends FunctionalComponent<Props>,
  Props
>(
  originalComponent: TestedComponent,
  options?: MountingOptions<Props>
): VueWrapper<ComponentPublicInstance<Props>>
```